static/app/utils/profiling/renderers/textRenderer.tsx

```diff
@@ -126,19 +131,49 @@ class TextRenderer {
       // Offset x by 1x the padding
       const x = frameInPhysicalSpace[0] + HALF_SIDE_PADDING;
 
-      this.context.fillText(
-        trimTextCenter(
-          frame.frame.name,
-          findRangeBinarySearch(
-            {low: 0, high: paddedRectangleWidth},
-            n => this.measureAndCacheText(frame.frame.name.substring(0, n)),
-            paddedRectangleWidth
-          )[0]
-        ),
-        x,
-        y
+      const frameName = frame.frame.name;
+      const trim = trimTextCenter(
+        frameName,
+        findRangeBinarySearch(
+          {low: 0, high: paddedRectangleWidth},
+          n => this.measureAndCacheText(frame.frame.name.substring(0, n)).width,
+          paddedRectangleWidth
+        )[0]
       );
 
+      const {text: trimText} = trim;
+
+      if (flamegraphSearch) {
+        const searchResults = flamegraphSearch.results ?? {};
+        const searchQuery = flamegraphSearch.query;
+        const frameId = getFlamegraphFrameSearchId(frame);
+        const isFrameInSearchResults = searchResults[frameId];
+
+        if (isFrameInSearchResults && searchQuery) {
+          const re = new RegExp(searchQuery, 'ig');
+
+          for (const highlightedBounds of matchHighlightedBounds(frameName, re, trim)) {
+            const [startIndex, endIndex] = highlightedBounds;
+            const highlightTextSize = this.measureAndCacheText(
+              trimText.substring(startIndex, endIndex)
+            );
+            const frontMatter = trimText.slice(0, startIndex);
+            const startHighlightX = this.measureAndCacheText(frontMatter).width;
+            this.context.fillStyle = '#FFFF00';
+            this.context.fillRect(
+              startHighlightX + x,
+              y - highlightTextSize.fontBoundingBoxAscent,
+              highlightTextSize.width,
+              highlightTextSize.fontBoundingBoxAscent +
+                highlightTextSize.fontBoundingBoxDescent
+            );
+          }
+        }
+      }
```

Comment:
This is making an assumption that the search is always a regex search. However, we currently have 2 types of searches
- regex
- fuse (check out `flamegraphSearch` in `static/app/components/profiling/flamegraphSearch.tsx`)

We should probably focus our efforts on fuse for the time being and drop support for regex until there's a need for it as supporting both is making things more complicated than it needs to be. As for fuse, it has an `includeMatches` options that we can use to find the characters that were matched.

Reply:
there's only 1 issue with this currently; fuse is giving us some fuzzy matches and yields some extraneous indices. we save on the regexp computation, but result in more indices to loop over.

for example, searching for `runloo` would result in the following:

![image](https://user-images.githubusercontent.com/7349258/174115154-52fc1547-d73f-4403-9a5a-1ad77986afce.png)


not sure what's better here. we can certainly fine tune our `Fuse` instance. i've noticed while developing we're getting `rects` highlighted as orange for matches, even though we don't actually have a match.

Reply:
While I agree that it's not great that it's showing matches else where but that's because the actual match wasn't on the exact substring but a fuzzy search across those indices indicated. So there's a decent chance that we'll have search results where you don't highlight anything because you're not highlighting what fuse actually used to determine the match.

We should sync with @JonasBa and decide what the behaviour we want here is.

Reply:
> So there's a decent chance that we'll have search results where you don't highlight anything because you're not highlighting what fuse actually used to determine the match.

I think you are correct here @Zylphrex. 

My general sentiment is that whichever piece of code marks a result as a match should also be responsible for giving us the indices of why it matched, else we end up with out of sync highlighting and potentionally confusing experience.

For the case where fuse generates multiple indices, we are probably ok by selecting the one (or many) by max distance as that distance is a good proxy for relevance.

Considering the following cases: 
- indices.length > 1 like `[[0,0], [0,3]]` we take max and just draw `[0,3]`
- indices.length > 1 and we have multiple matches `[[0,0], [0,3], [4,7]]` we draw `[0,3] and [4,7]` as they both could be equally important.
- I expect the above two conditions to eliminate the occurrence where we have many different single character matches (as there should always be a max higher than those two and thus they get discarded)

Reply:
i agree it makes more sense to handle capturing the indices from Fuse and in general where we do the actual search operation.

we _are_ technically missing some frames, however we highlight some potentially irrelevant frames aswell. here is an example of searching for `"display"` and i yield a match for `"_dispatch_main_queue_callback_4CF"`

![image](https://user-images.githubusercontent.com/7349258/174355616-05397d12-f558-49d7-b755-c4c48d6c2407.png)


i suppose its a bit subjective to the user and what we want our outcome to be, but when i think of searching for something that's more technical in nature i would prefer a more exact match rather than a fuzzy match. it's like trying to find something in our codebase with vscode; there is no tolerance for fuzzy matching its always exact w/ a few modifying options or regex.

i'll work on having the indices propagate from search to our draw method for now, and we can figure out the rest later

Reply:
>it's like trying to find something in our codebase with vscode; there is no tolerance for fuzzy matching its always exact w/ a few modifying options or regex.

I agree, we shouldn't be fuzzy matching too many things. Have you tried lowering the threshold we use for the index?

> i'll work on having the indices propagate from search to our draw method for now, and we can figure out the rest late

🙌🏼 sounds great. Feel free to break the PR into the propagation of indices and later on adding the actual highlighting if you think it makes sense